com.chartboost.helium/Android~/helium-android-bridge/src/main/java/com/chartboost/heliumsdk/unity/HeliumUnityBridge.kt

```diff
@@ -110,267 +107,138 @@ class HeliumUnityBridge {
+                    HeliumEventConsumerWithError { placementName: String, error: String ->
+                        interstitialEventsListener?.DidCloseInterstitial(placementName, error)
+                    })
+            }
 
-                    override fun didClose(placementName: String, error: HeliumAdError?) {
-                        serializeHeliumEvent(
-                            placementName,
-                            error,
-                            HeliumEventConsumer { placementName: String, errorCode: Int, errorDescription: String ->
-                                interstitialEventsListener?.DidCloseInterstitial(
-                                    placementName,
-                                    errorCode,
-                                    errorDescription
-                                )
-                            })
-                    }
+            override fun onAdClicked(placementName: String) {
+                serializeHeliumEvent(placementName,
+                    HeliumEventConsumer { placementName: String ->
+                        interstitialEventsListener?.DidClickInterstitial(placementName)
+                    })
+            }
 
-                    override fun didClick(placementName: String, error: HeliumAdError?) {
-                        serializeHeliumEvent(
-                            placementName,
-                            error,
-                            HeliumEventConsumer { placementName: String, errorCode: Int, errorDescription: String ->
-                                interstitialEventsListener?.DidClickInterstitial(
-                                    placementName,
-                                    errorCode,
-                                    errorDescription
-                                )
-                            })
-                    }
+            override fun onAdImpressionRecorded(placementName: String) {
+                serializeHeliumEvent(placementName,
+                    HeliumEventConsumer { placementName: String ->
+                        interstitialEventsListener?.DidRecordImpression(placementName)
+                    })
+            }
 
-                    override fun didRecordImpression(placementName: String) {
-                        serializeHeliumEvent(
-                            placementName,
-                            null,
-                            HeliumEventConsumer { placementName: String, errorCode: Int, errorDescription: String ->
-                                interstitialEventsListener?.DidRecordImpression(
-                                    placementName,
-                                    errorCode,
-                                    errorDescription
-                                )
-                            })
-                    }
-                })
-        )
+            override fun onAdRewarded(placementName: String) {
+                TODO("Not yet implemented")
```

Comment:
this may bork on runtime... have wrapped as a comment for now

Reply:
Actually... this should not trigger for interstitial, we can add a warning log in the case it does.

Reply:
Right, but this will crash a publisher's app if it ever does trigger the callback. Which, we do not want.

Reply:
this may start happening ;)
but it should come with a new unity version so /shrug